### PR TITLE
monitor a single file entity

### LINKED ISSUE #24: Add support for monitoring the update of one entity
Add logic to `monitor._find_modified_entities_file`

### LINKED ISSUE #42: Add support to monitor a Synapse Table entity

### PR DESCRIPTION
**Pull request by hhunterzinck**
Fixes #24 
Fixes #42 

- Monitors a single file entity
- Removes File from not implemented list in tests
- Add Schema in addition to File as valid single entity types for monitoring

### CODE REVIEW

**Comment by thomasyu888** [comment]
@hhunterzinck would you like me to add the tests for the monitor single entity?

**Comment by thomasyu888** [comment]
One case I did not think about is: Did we want to ever monitor if the fileview entity is modified?

File: synapsemonitor/monitor.py [hunk 1]
```diff
@@ -100,7 +102,14 @@ def _find_modified_entities_file(syn: Synapse, syn_id: str, days: int = 1) -> li
     Returns:
         List of synapse ids
     """
-    raise NotImplementedError("Files not supported yet")
```
Comment (thomasyu888) [inline synapsemonitor/monitor.py:103]: I wonder if this function should be called `_find_modified_entities_single_entity`?

File: synapsemonitor/monitor.py [hunk 2]
```diff
@@ -100,7 +102,14 @@ def _find_modified_entities_file(syn: Synapse, syn_id: str, days: int = 1) -> li
...
+    entity = syn.get(syn_id, downloadFile=False)
+    utc_mod = datetime.strptime(entity["modifiedOn"], "%Y-%m-%dT%H:%M:%S.%fZ")
+    local_mod = utc_mod.replace(tzinfo=tz.tzutc()).astimezone(tz.tzlocal())
+    local_now = datetime.now().replace(tzinfo=tz.tzlocal())
```
Comment (thomasyu888) [inline synapsemonitor/monitor.py:108]: If i'm reading this correctly, are you modifying the entity modified on date to local time then getting the current local time?  Can we just get the current UTC time and subtract?

**Comment by hhunterzinck** [comment]
> 

We probably need to distinguish between monitoring an entity itself and the contents of the entity.  Then we could just treat all entities the same and only have to provide custom functionality for looking at the contents (e.g. fileview, folder/project contents).  Perhaps another flag (--contents) to activate the monitoring of the contents?  Then it would need to distinguish between entity types

File: synapsemonitor/monitor.py (hunk 2 above)
Comment (hhunterzinck) [reply synapsemonitor/monitor.py]: yes!  nice catch.  made the fix

**Comment by thomasyu888** [comment]
@hhunterzinck That is a great idea - we can open a GitHub issue to track.

**Comment by hhunterzinck** [comment]
Two existing issues that should probably be addressed in order:
Issue #23
Issue #45 
Maybe merging changes from this branch first.  Then implement container contents monitoring.  Then implement flag for generic single entity monitoring versus contents monitoring.

**Comment by thomasyu888** [comment]
Ya let's merge this first then work on other issues.

**Pull request merged**